pkg/kubelet/pod_workers.go

```diff
@@ -64,29 +72,33 @@ func (p *podWorkers) managePodLoop(podUpdates <-chan api.BoundPod) {
 			glog.Errorf("Error listing containers while syncing pod: %v", err)
 			continue
 		}
-		err = p.syncPodFun(&newPod, containers)
+		err = p.syncPodFun(newWork.pod, containers)
 		if err != nil {
-			glog.Errorf("Error syncing pod %s, skipping: %v", newPod.UID, err)
-			record.Eventf(&newPod, "failedSync", "Error syncing pod, skipping: %v", err)
+			glog.Errorf("Error syncing pod %s, skipping: %v", newWork.pod.UID, err)
+			record.Eventf(newWork.pod, "failedSync", "Error syncing pod, skipping: %v", err)
 			continue
 		}
 	}
 }
 
-func (p *podWorkers) UpdatePod(pod api.BoundPod) {
+// Apply the new setting to the specified pod. updateComplete is called when the update is completed.
+func (p *podWorkers) UpdatePod(pod *api.BoundPod, updateComplete func()) {
 	uid := pod.UID
-	var podUpdates chan api.BoundPod
+	var podUpdates chan workUpdate
 	var exists bool
 
 	p.podLock.Lock()
 	defer p.podLock.Unlock()
 	if podUpdates, exists = p.podUpdates[uid]; !exists {
 		// TODO(wojtek-t): Adjust the size of the buffer in this channel
-		podUpdates = make(chan api.BoundPod, 5)
+		podUpdates = make(chan workUpdate, 5)
 		p.podUpdates[uid] = podUpdates
 		go p.managePodLoop(podUpdates)
 	}
-	podUpdates <- pod
+	podUpdates <- workUpdate{
+		pod:               pod,
+		updateCompleteFun: updateComplete,
```

Comment:
We have Fn, Fun, and Func. I wouldn't say no to Fun.
You can refactor in a separate PR if you want. Let's get this one in.